src/app/reporting/Engine.cpp

```diff
@@ -176,6 +196,10 @@ DataModel::ActionReturnStatus RetrieveClusterData(DataModel::Provider * dataMode
         // the are routed through metadata.
         status = ReadGlobalAttributeFromMetadata(dataModel, readRequest.path, attributeValueEncoder);
     }
+    else if (auto existence_status = ValidateAttributeIsReadable(dataModel, path); existence_status.has_value())
```

Comment:
"existence_status" is a bad name: the attribute may exist but be write-only, no?